python/ola/MACAddress.py

```diff
@@ -51,19 +51,44 @@ def __hash__(self):
   def __repr__(self):
     return self.__str__()
 
-  def __cmp__(self, other):
-    if other is None:
-      return 1
-    return cmp(self.mac_address, other.mac_address)
+  def __eq__(self, other):
+    if other.__class__ is not self.__class__:
```

Comment:
Do we need to do:
`if other is None or other.__class__ is not self.__class__:`

To protect against other being None, or does None still have __class__?

Reply:
On the topic of `None.__class__`, anecdotally it has `__class__`:

```
$ python2
Python 2.7.15+ (default, Nov 27 2018, 23:36:35) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> None.__class__
<type 'NoneType'>
```

```
$ python3
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> None.__class__
<class 'NoneType'>
```

`isinstance(other, self.__class__)` would take care of this for you.

If `isinstance` isn't desired, it seems like a more bulletproof approach might be to wrap these comparisons in a `try` `except AttributeError` block, because it is ambiguous in [the documentation of `__class__`](https://docs.python.org/3/library/stdtypes.html#special-attributes) which instances have `__class__`.  In Python 3 a value not having a `__class__` attribute is an even more distant edge case.

Is there some reason this doesn't this return `NotImplemented` like the other operators?

Reply:
It doesn't return `NotImplemented` because if classes don't match, the two objects aren't equals.

Reply:
Do you want to switch to isinstance @nils-van-zuijlen or is there some reason not to do so?